modules/launcher/src/main/java/org/mule/module/launcher/artifact/DefaultResourceReleaser.java

```diff
@@ -32,6 +34,37 @@
     public void release()
     {
         deregisterJdbcDrivers();
+
+        cleanUpResourceBundle();
+    }
+
+    private void cleanUpResourceBundle()
+    {
+        try
+        {
+            ResourceBundle.clearCache(this.getClass().getClassLoader());
+        }
+        catch(Exception e)
+        {
+            logger.warn("Couldn't clean up ResourceBundle", e);
+        }
+
+        try
+        {
+            // NONEXISTENT_BUNDLE leaves a cacheKey referencing to this classloader
+            Field nonExistentBundleField = ResourceBundle.class.getDeclaredField("NONEXISTENT_BUNDLE");
```

Comment:
I don't get why this is required if the cache was cleaned up previously. Can you add a comment explaining the reasons?